src/Farmer/Common.fs

```diff
@@ -677,7 +680,9 @@ module ContainerGroup =
         | PublicAddressWithDns of DnsName:string
         | PrivateAddress
     /// A secret file which will be encoded as base64 and attached to a container group.
-    type SecretFile = SecretFile of Name:string * Secret:byte array
+    type SecretFile =
+            | SecretFile of Name:string * Secret:byte array
```

Comment:
I think that we should consider renaming either the one (or both) of the cases or the overall type - something doesn't feel right here. Either it's a `SecretFile` and the first case needs renaming, or the first case is a `SecretFile` and the top type name should be something different.

Or something else entirely :-)

Reply:
That makes sense - thanks for explaining this.

Reply:
In the keyvault version, what would be stored in the secret - the UTF8-encoded byte array?

Reply:
Is the parameter directly designed for KeyVault (in which case we can supply something like a KeyVault ref) or is it simply a parameter that could be used to supply the encoded data directly as an argument?

Reply:
I'm now thinking that perhaps we shouldn't allow people to supply the secret data directly in the template and *force* it to be a parameter!